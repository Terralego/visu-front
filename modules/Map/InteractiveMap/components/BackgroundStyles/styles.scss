@import "node_modules/@blueprintjs/core/lib/scss/variables";

.radioGroup {
  padding: 10px;
  line-height: 2em;
}

.bp3-popover-content {
  padding: 4px;
}

.bgLayer-radio {
  padding: 8px 36px 8px 8px;
  border: 1px solid #182026;
  margin: 0;
  &:not(:last-of-type) {
    border-bottom: 0px;
  }
}

.bp3-control {
  &:not(.bp3-align-right) {
    .bp3-control-indicator {
      margin-left: -12px;
    }
  }
}
.mapboxgl-ctrl-background-styles {
  position: relative;

  .bp3-popover-wrapper {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    line-height: 30px;
  }
}
