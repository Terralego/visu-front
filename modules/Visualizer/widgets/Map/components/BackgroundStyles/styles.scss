.radioGroup {
  padding: 10px;
  line-height: 2em;
}

.popoverPos {
  position: absolute;
  right: 10px;
  top: 110px;
  z-index: 10;
}