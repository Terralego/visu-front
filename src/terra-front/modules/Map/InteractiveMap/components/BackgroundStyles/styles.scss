@import "node_modules/@blueprintjs/core/lib/scss/variables";
@import '../../../../../utils/variables.scss';

.radioGroup {
  padding: .5rem;
  line-height: 2em;
}

.bp4-popover-content {
  padding: .25rem;
}

.bgLayer-radio {
  padding: .5rem 2.25rem .5rem .5rem;
  margin: 0;
  border: 1px solid #f0f0f0;
  .bp4-dark & {
    border-color: #182026;
  }
  &:not(:last-of-type) {
    border-bottom: 0;
  }
}

.bp4-control {
  &:not(.bp4-align-right) {
    .bp4-control-indicator {
      margin-left: -12px;
    }
  }
}

@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .bp4-popover-content .radioGroup {
    max-width: 15.5rem;
  }
}
