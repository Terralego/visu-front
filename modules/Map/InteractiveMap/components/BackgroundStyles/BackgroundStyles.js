import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { Radio, RadioGroup, Button, Icon, Popover } from '@blueprintjs/core';

import './styles.scss';

export class BackgroundStyles extends React.Component {
  static propTypes = {
    styles: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
    })).isRequired,
    selected: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
  }

  state = {
    showRadioGroup: false,
  }

  onAdd (map) {
    this.map = map;
    this.container = document.createElement('div');
    this.container.className = 'mapboxgl-ctrl mapboxgl-ctrl-group mapboxgl-ctrl-search';
    ReactDOM.render(
      <BackgroundStyles
        container={this.container}
        {...this.props}
      />,
      this.container,
    );
    return this.container;
  }

  onRemove () {
    this.container.parentNode.removeChild(this.container);
    delete this.map;
  }

  toggleSelector = () => {
    const { showRadioGroup } = this.state;
    this.setState({ showRadioGroup: !showRadioGroup });
  }

  onChange = ({ target: { value } }) => {
    const { onChange } = this.props;
    onChange(value);
  }

  render () {
    const { styles, selected } = this.props;
    const { toggleSelector, onChange } = this;

    return (
      <Popover
        className="bp3-dark popoverPos"
        content={(
          <div className="radioGroup">
            <RadioGroup
              label="Fond de carte"
              onChange={onChange}
              selectedValue={selected}
            >
              {styles.map(({ label, url }) => (
                <Radio
                  className="bgLayer-radio"
                  key={`${label}${url}`}
                  label={label}
                  value={url}
                />
              ))}
            </RadioGroup>
          </div>
        )}
      >
        <Button onClick={toggleSelector}>
          {<Icon icon="layers" />}
        </Button>
      </Popover>
    );
  }
}

export default BackgroundStyles;
