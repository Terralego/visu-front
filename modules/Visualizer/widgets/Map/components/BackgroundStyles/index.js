import React from 'react';
import { Radio, RadioGroup, Button, Icon, Popover } from '@blueprintjs/core';

import './styles.scss';

export class BackgroundStyles extends React.Component {
  state = {
    showRadioGroup: false,
  }

  toggleSelector = () => {
    const { showRadioGroup } = this.state;
    this.setState({ showRadioGroup: !showRadioGroup });
  }

  onChange = ({ target: { value } }) => {
    const { onChange } = this.props;
    onChange(value);
  }

  render () {
    const { styles, selected } = this.props;
    const { toggleSelector, onChange } = this;

    return (
      <Popover
        className="bp3-dark popoverPos"
        content={(
          <div className="radioGroup">
            <RadioGroup
              label="Fond de carte"
              onChange={onChange}
              selectedValue={selected}
            >
              {styles.map(({ label, url }) => (
                <Radio label={label} value={url} />
              ))}
            </RadioGroup>
          </div>
        )}
      >
        <Button onClick={toggleSelector}>
          {<Icon icon="page-layout" />}
        </Button>
      </Popover>
    );
  }
}

export default BackgroundStyles;
