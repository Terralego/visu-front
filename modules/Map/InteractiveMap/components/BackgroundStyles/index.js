import BackgroundStyles from './BackgroundStyles';

export * from './BackgroundStyles';
export default BackgroundStyles;
