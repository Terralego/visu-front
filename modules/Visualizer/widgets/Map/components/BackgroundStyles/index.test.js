import React from 'react';
import renderer from 'react-test-renderer';
import BackgroundStyles from '.';

it('should render correctly', () => {
  const style = [{ label: 'Onizuka', url: 'Sensei' }];
  const tree = renderer.create(<BackgroundStyles selected="Sensei" styles={style} />).toJSON();
  expect(tree).toMatchSnapshot();
});
