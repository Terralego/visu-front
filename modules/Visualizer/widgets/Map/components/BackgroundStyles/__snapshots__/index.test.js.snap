// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<span
  className="bp3-popover-wrapper bp3-dark popoverPos"
>
  <span
    className="bp3-popover-target"
    onClick={[Function]}
  >
    <button
      className="bp3-button"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-button-text"
      >
        <span
          className="bp3-icon bp3-icon-page-layout"
          icon="page-layout"
        >
          <svg
            data-icon="page-layout"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              page-layout
            </desc>
            <path
              d="M15 .95H1c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-12c0-.55-.45-1-1-1zm-9 12H2v-6h4v6zm8 0H7v-6h7v6zm0-7H2v-3h12v3z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </span>
    </button>
  </span>
</span>
`;
